mcs/class/System/System.Net/HttpListenerRequest.cs

```diff
@@ -403,7 +403,7 @@ internal bool FlushInput ()
 		}
 
 		public bool IsLocal {
-			get { return IPAddress.IsLoopback (RemoteEndPoint.Address); }
+			get { return LocalEndPoint.Address.Equals (RemoteEndPoint.Address); }
```

Comment:
I'm not sure this is a complete solution. If the device has more than one interface it is possible that the connection will be between two different addresses which are still local to the machine, however. I think that RemoteEndPoint.Address has to be checked against the list of interfaces on the local machine.